### PR TITLE
[Docs] Consolidate install partials

### LINKED ISSUE #18606: `install-linux.mdx` can cause trouble for Cloud users
## Details
If Teleport Cloud happens to be one major version behind the latest self-hosted release, `install-linux.mdx` will result in installing a Teleport service that refuses to run. We should always include `install-agent-cloud.mdx` for Cloud users, since it shows how to install a Cloud-compatible Teleport version.

**As an alternative:** We can edit `install-linux.mdx` to include Cloud-compatible instructions and remove `install-agent-cloud.mdx`.

## Applies To
Pages that mention `install-linux.mdx` but not `install-agent-cloud.mdx`:

```bash
$ find docs/pages -name "*.mdx" -exec awk 'BEGIN{i=0;c=0;} /install-linux\.mdx/{i=1} /install-agent-cloud\.mdx/{c=1} END{if (i==1 && c==0) print FILENAME}' {} \;
docs/pages/database-access/getting-started.mdx
docs/pages/database-access/guides/mongodb-atlas.mdx
docs/pages/database-access/guides/cockroachdb-self-hosted.mdx
docs/pages/database-access/guides/elastic.mdx
docs/pages/database-access/guides/sql-server-ad.mdx
docs/pages/database-access/guides/dynamodb.mdx
docs/pages/database-access/guides/snowflake.mdx
docs/pages/database-access/guides/redis-aws.mdx
docs/pages/database-access/guides/aws-cassandra-keyspaces.mdx
docs/pages/database-access/guides/mysql-self-hosted.mdx
docs/pages/database-access/guides/postgres-redshift.mdx
docs/pages/database-access/guides/mongodb-self-hosted.mdx
docs/pages/database-access/guides/postgres-cloudsql.mdx
docs/pages/database-access/guides/rds.mdx
docs/pages/database-access/guides/cassandra-self-hosted.mdx
docs/pages/database-access/guides/redis.mdx
docs/pages/database-access/guides/azure-postgres-mysql.mdx
docs/pages/database-access/guides/redis-cluster.mdx
docs/pages/database-access/guides/postgres-self-hosted.mdx
docs/pages/database-access/guides/mysql-cloudsql.mdx
docs/pages/database-access/guides/azure-redis.mdx
docs/pages/machine-id/guides/host-certificate.mdx
docs/pages/deploy-a-cluster/open-source.mdx
docs/pages/deploy-a-cluster/teleport-enterprise/getting-started.mdx
docs/pages/desktop-access/manual-setup.mdx
docs/pages/kubernetes-access/guides/standalone-teleport.mdx
docs/pages/kubernetes-access/guides/dynamic-registration.mdx
docs/pages/management/admin/daemon.mdx
docs/pages/management/admin/labels.mdx
docs/pages/management/admin/upgrading-the-teleport-binary.mdx
docs/pages/installation.mdx
docs/pages/application-access/getting-started.mdx
docs/pages/application-access/guides/connecting-apps.mdx
docs/pages/application-access/guides/tcp.mdx
```

In contrast, these pages mention both partials:

```bash
$ find docs/pages -name "*.mdx" -exec awk 'BEGIN{i=0;c=0;} /install-linux\.mdx/{i=1} /install-agent-cloud\.mdx/{c=1} END{if (i==1 && c==1) print FILENAME}' {} \;
docs/pages/management/admin/adding-nodes.mdx
docs/pages/server-access/getting-started.mdx
docs/pages/server-access/guides/bpf-session-recording.mdx
```

### PR DESCRIPTION
**Pull request by alexfornuto**
Following up #17498, while that PR was awaiting review other PRs came and went. One of them introduced a second partial covering installation of Teleport on nodes that would join a Teleport Cloud cluster.

This PR consolidates the useful additions from that partial into the main `install-linux.mdx` file, and adjusts the usage throughout the docs to use the newly consolidated partial.

Note to future self: One of the adjusted files is not backported to 11, so manual backporting will be required.

### CODE REVIEW

**Review by ptgott — APPROVED** [review]
Thanks for this!

File: docs/pages/includes/install-linux.mdx [hunk 1]
```diff
@@ -181,5 +181,14 @@ Next, use the appropriate commands for your environment to install your package.
 
   </TabItem>
   </Tabs>
+  <Details title="Is my Teleport instance compatible with Teleport Cloud?">
+
+  Before installing a `teleport` binary with a version besides v(=cloud.version=),
```
Comment (zmb3) [inline docs/pages/includes/install-linux.mdx:186]: How about we use `cloud.major_version` instead? There's nothing wrong with not being on the exact same patch.

**Review by zmb3 — APPROVED** [review]
Approving so as not to block you from merging.

Please consider my suggestion :-)

File: docs/pages/includes/install-linux.mdx (hunk 1 above)
Comment (alexfornuto) [reply docs/pages/includes/install-linux.mdx:186]:
```suggestion
  Before installing a `teleport` binary with a version besides v(=cloud.major_version=),
```

**Pull request merged**